Code/Editor/EditorFramework/Assets/Implementation/AssetUpdates.cpp

```diff
@@ -208,31 +212,38 @@ ezResult ezAssetCurator::EnsureAssetInfoUpdated(ezStringView sAbsFilePath, const
     // OVERWRITTEN
     SetAssetExistanceState(*m_KnownAssets[oldGuid], ezAssetExistanceState::FileRemoved);
     RemoveAssetTransformState(oldGuid);
+    newExistanceState = ezAssetExistanceState::FileAdded;
   }
 
-  bool bNewAsset = false;
   if (pCurrentAssetInfo)
   {
     UntrackDependencies(pCurrentAssetInfo);
     pCurrentAssetInfo->Update(pNewAssetInfo);
+    // Only update if it was not already set to not overwrite, e.g. FileMoved.
+    if (newExistanceState == ezAssetExistanceState::FileUnchanged)
+      newExistanceState = ezAssetExistanceState::FileModified;
   }
   else
   {
-    bNewAsset = true;
     pCurrentAssetInfo = pNewAssetInfo.Release();
     m_KnownAssets[newGuid] = pCurrentAssetInfo;
+    newExistanceState = ezAssetExistanceState::FileAdded;
   }
 
   TrackDependencies(pCurrentAssetInfo);
   CheckForCircularDependencies(pCurrentAssetInfo).IgnoreResult();
   UpdateAssetTransformState(newGuid, ezAssetInfo::TransformState::Unknown);
   // Don't call SetAssetExistanceState on newly created assets as their data structure is initialized in UpdateSubAssets for the first time.
-  if (!bNewAsset)
-    SetAssetExistanceState(*pCurrentAssetInfo, ezAssetExistanceState::FileModified);
+  if (newExistanceState != ezAssetExistanceState::FileAdded)
+    SetAssetExistanceState(*pCurrentAssetInfo, newExistanceState);
   UpdateSubAssets(*pCurrentAssetInfo);
 
   InvalidateAssetTransformState(newGuid);
-
+  if (pFiles->LinkDocument(sAbsFilePath, pCurrentAssetInfo->m_Info->m_DocumentID).Failed())
+  {
+    // #TODO_ASSET
```

Comment:
still todo ?

Reply:
Probably not necessary anymore. This can only fail of the file is no longer part of the model, but as we are in all callback for this specific file, this shouldn't be possible.